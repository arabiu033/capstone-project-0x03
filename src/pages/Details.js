import React from 'react';
import { useSelector } from 'react-redux';
import Spinner from '../components/Spinner';
import Card from '../components/Card';

const Details = () => {
  const { laureates, loading } = useSelector((state) => ({ ...state.laureates }));
  if (loading === 'Loading Api') return (<Spinner />);
  return (
    <div className="laureates">
      <Card laureates={laureates} />
    </div>
  );
};

export default Details;
