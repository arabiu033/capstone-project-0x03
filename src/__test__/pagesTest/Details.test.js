import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/configureStore';

import Details from '../../pages/Details';

test('renders correctly Header component', () => {
  const det = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <Details />
      </Provider>
    </BrowserRouter>
  ).toJSON();
  expect(det).toMatchSnapshot();
});