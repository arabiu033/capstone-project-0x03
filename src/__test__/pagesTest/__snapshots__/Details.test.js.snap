// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Header component 1`] = `
<div
  className="spinner"
>
  <svg
    aria-busy={true}
    aria-label="dna-loading"
    className=""
    data-testid="dna-svg"
    height="80"
    preserveAspectRatio="xMidYMid"
    role="status"
    style={Object {}}
    viewBox="0 0 100 100"
    width="80"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <circle
      cx="6.451612903225806"
      cy="60.6229"
      fill="rgba(233, 12, 89, 0.5125806451612902)"
      r="3.41988"
    >
      <animate
        attributeName="r"
        begin="-0.5s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="0s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-0.5s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="rgba(233, 12, 89, 0.5125806451612902);#ff0033;rgba(233, 12, 89, 0.5125806451612902)"
      />
    </circle>
    <circle
      cx="6.451612903225806"
      cy="39.3771"
      fill="#46dff0"
      r="2.58012"
    >
      <animate
        attributeName="r"
        begin="-1.5s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-1s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-0.5s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="#46dff0;rgba(53, 58, 57, 0.1435483870967742);#46dff0"
      />
    </circle>
    <circle
      cx="16.129032258064512"
      cy="68.1552"
      fill="rgba(233, 12, 89, 0.5125806451612902)"
      r="3.17988"
    >
      <animate
        attributeName="r"
        begin="-0.7s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-0.2s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-0.7s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="rgba(233, 12, 89, 0.5125806451612902);#ff0033;rgba(233, 12, 89, 0.5125806451612902)"
      />
    </circle>
    <circle
      cx="16.129032258064512"
      cy="31.8448"
      fill="#46dff0"
      r="2.82012"
    >
      <animate
        attributeName="r"
        begin="-1.7s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-1.2s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-0.7s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="#46dff0;rgba(53, 58, 57, 0.1435483870967742);#46dff0"
      />
    </circle>
    <circle
      cx="25.806451612903224"
      cy="69.3634"
      fill="rgba(233, 12, 89, 0.5125806451612902)"
      r="2.93988"
    >
      <animate
        attributeName="r"
        begin="-0.9s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-0.4s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-0.9s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="rgba(233, 12, 89, 0.5125806451612902);#ff0033;rgba(233, 12, 89, 0.5125806451612902)"
      />
    </circle>
    <circle
      cx="25.806451612903224"
      cy="30.6366"
      fill="#46dff0"
      r="3.06012"
    >
      <animate
        attributeName="r"
        begin="-1.9s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-1.4s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-0.9s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="#46dff0;rgba(53, 58, 57, 0.1435483870967742);#46dff0"
      />
    </circle>
    <circle
      cx="35.48387096774193"
      cy="65.3666"
      fill="rgba(233, 12, 89, 0.5125806451612902)"
      r="2.69988"
    >
      <animate
        attributeName="r"
        begin="-1.1s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-0.6s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-1.1s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="rgba(233, 12, 89, 0.5125806451612902);#ff0033;rgba(233, 12, 89, 0.5125806451612902)"
      />
    </circle>
    <circle
      cx="35.48387096774193"
      cy="34.6334"
      fill="#46dff0"
      r="3.30012"
    >
      <animate
        attributeName="r"
        begin="-2.1s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-1.6s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-1.1s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="#46dff0;rgba(53, 58, 57, 0.1435483870967742);#46dff0"
      />
    </circle>
    <circle
      cx="45.16129032258064"
      cy="53.8474"
      fill="rgba(233, 12, 89, 0.5125806451612902)"
      r="2.45988"
    >
      <animate
        attributeName="r"
        begin="-1.3s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-0.8s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-1.3s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="rgba(233, 12, 89, 0.5125806451612902);#ff0033;rgba(233, 12, 89, 0.5125806451612902)"
      />
    </circle>
    <circle
      cx="45.16129032258064"
      cy="46.1526"
      fill="#46dff0"
      r="3.54012"
    >
      <animate
        attributeName="r"
        begin="-2.3s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-1.8s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-1.3s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="#46dff0;rgba(53, 58, 57, 0.1435483870967742);#46dff0"
      />
    </circle>
    <circle
      cx="54.838709677419345"
      cy="39.3771"
      fill="rgba(233, 12, 89, 0.5125806451612902)"
      r="2.58012"
    >
      <animate
        attributeName="r"
        begin="-1.5s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-1s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-1.5s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="rgba(233, 12, 89, 0.5125806451612902);#ff0033;rgba(233, 12, 89, 0.5125806451612902)"
      />
    </circle>
    <circle
      cx="54.838709677419345"
      cy="60.6229"
      fill="#46dff0"
      r="3.41988"
    >
      <animate
        attributeName="r"
        begin="-2.5s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-2s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-1.5s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="#46dff0;rgba(53, 58, 57, 0.1435483870967742);#46dff0"
      />
    </circle>
    <circle
      cx="64.51612903225805"
      cy="31.8448"
      fill="rgba(233, 12, 89, 0.5125806451612902)"
      r="2.82012"
    >
      <animate
        attributeName="r"
        begin="-1.7s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-1.2s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-1.7s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="rgba(233, 12, 89, 0.5125806451612902);#ff0033;rgba(233, 12, 89, 0.5125806451612902)"
      />
    </circle>
    <circle
      cx="64.51612903225805"
      cy="68.1552"
      fill="#46dff0"
      r="3.17988"
    >
      <animate
        attributeName="r"
        begin="-2.7s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-2.2s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-1.7s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="#46dff0;rgba(53, 58, 57, 0.1435483870967742);#46dff0"
      />
    </circle>
    <circle
      cx="74.19354838709677"
      cy="30.6366"
      fill="rgba(233, 12, 89, 0.5125806451612902)"
      r="3.06012"
    >
      <animate
        attributeName="r"
        begin="-1.9s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-1.4s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-1.9s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="rgba(233, 12, 89, 0.5125806451612902);#ff0033;rgba(233, 12, 89, 0.5125806451612902)"
      />
    </circle>
    <circle
      cx="74.19354838709677"
      cy="69.3634"
      fill="#46dff0"
      r="2.93988"
    >
      <animate
        attributeName="r"
        begin="-2.9s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-2.4s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-1.9s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="#46dff0;rgba(53, 58, 57, 0.1435483870967742);#46dff0"
      />
    </circle>
    <circle
      cx="83.87096774193547"
      cy="34.6334"
      fill="rgba(233, 12, 89, 0.5125806451612902)"
      r="3.30012"
    >
      <animate
        attributeName="r"
        begin="-2.1s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-1.6s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-2.1s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="rgba(233, 12, 89, 0.5125806451612902);#ff0033;rgba(233, 12, 89, 0.5125806451612902)"
      />
    </circle>
    <circle
      cx="83.87096774193547"
      cy="65.3666"
      fill="#46dff0"
      r="2.69988"
    >
      <animate
        attributeName="r"
        begin="-3.1s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-2.6s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-2.1s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="#46dff0;rgba(53, 58, 57, 0.1435483870967742);#46dff0"
      />
    </circle>
    <circle
      cx="93.54838709677418"
      cy="46.1526"
      fill="rgba(233, 12, 89, 0.5125806451612902)"
      r="3.54012"
    >
      <animate
        attributeName="r"
        begin="-2.3s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-1.8s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-2.3s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="rgba(233, 12, 89, 0.5125806451612902);#ff0033;rgba(233, 12, 89, 0.5125806451612902)"
      />
    </circle>
    <circle
      cx="93.54838709677418"
      cy="53.8474"
      fill="#46dff0"
      r="2.45988"
    >
      <animate
        attributeName="r"
        begin="-3.3s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="2.4000000000000004;3.5999999999999996;2.4000000000000004"
      />
      <animate
        attributeName="cy"
        begin="-2.8s"
        calcMode="spline"
        dur="2s"
        keySplines="0.5 0 0.5 1;0.5 0 0.5 1"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="30.5;69.5;30.5"
      />
      <animate
        attributeName="fill"
        begin="-2.3s"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="#46dff0;rgba(53, 58, 57, 0.1435483870967742);#46dff0"
      />
    </circle>
  </svg>
</div>
`;
